### PR TITLE
Add basic SARIF support

### PR DESCRIPTION
**Pull request by chao2zhang**
## Description
This addresses #1102. SARIF website: https://sarifweb.azurewebsites.net/
- This makes ktlint to depend on `https://github.com/detekt/sarif4k`, which uses kotlinx-serialization to generate the JSON. Schema makes the code safer and less error-prone. If we want to avoid additional dependency, I could change the code to yield JSON using basic string formatting and concatenation. 
- The level in SARIF report is currently set to ERROR. I believe this fits with the ktlint's strategy of no configuration.

## Checklist
- [X] tests are added
- [X] `CHANGELOG.md` is updated

## Follow-up
- [ ] `SarifSchema210.runs[0].tool.driver.rules` is empty for now. Ideally, this should be a list of rules that were active in the current run. I will address this in a follow-up PR since this requires refactoring.
- [ ] I have not yet added other information like autoCorrected or autoCorrectible in the output.
- [ ] Potentially we can also add github workflow setup in the documentation. There are other community projects that does this so this is optional to me.

## Test
- I ran `klint --reporter=sarif` on the `ktlint` itself. I also validated the output on `https://sarifweb.azurewebsites.net/Validation` to make sure that there is no basic warnings.

### CODE REVIEW

File: ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        val sarifSchema210 = SarifSchema210(
+            schema = "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
+            version = Version.The210,
+            runs = listOf(
+                Run(
+                    tool = Tool(
+                        driver = ToolComponent(
+                            downloadURI = "https://github.com/pinterest/ktlint/releases/tag/$version",
+                            fullName = "ktlint",
+                            guid = "7e96e861-6db7-44ea-a1e4-56bb4554ffe9",
```
Comment (romtsn) [inline ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt:78]: hm, where is this coming from?

File: ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        physicalLocation = PhysicalLocation(
+                            region = Region(
+                                startLine = err.line.toLong(),
+                                startColumn = err.col.toLong()
+                            ),
+                            artifactLocation = workingDirectory?.let { workingDirectory ->
+                                ArtifactLocation(
+                                    uri = File(file).relativeTo(workingDirectory).path,
+                                    uriBaseID = SRCROOT
+
```
Comment (romtsn) [inline ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt:54]:
```suggestion
```

**Review by romtsn — APPROVED** [review]
Don't know a thing about sarif, but I guess you know what you're doing 👍

File: ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt (hunk 1 above)
Comment (chao2zhang) [reply ktlint-reporter-sarif/src/main/kotlin/com/pinterest/ktlint/reporter/sarif/SarifReporter.kt:78]: guid is a unique identifier for the component: https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317539. I generated a random one for the time being.

It is optional, I can remove this.

**Comment by shashachu** [comment]
@chao2zhang apologies for the delay and thanks for the contribution. the CI build is now fixed so if you rebase I can go ahead and merge your change.

**Comment by chao2zhang** [comment]
Updated the PR, it looks like the PR is pending approval to kick off github actions

**Pull request merged**